### DISCUSSION TITLE
Azure AD Auth

### DISCUSSION THREAD

**Discussion by jemery27** [netbox-community/netbox#10510] [Q&A]
Kind of generic and specific question. First generic - we would like to start using the Azure AD auth - we have set it up and so far it works - still some questions about groups and permissions. But beyond that wondering about mapping existing users to Azure AD users specifically for change log purposes. Is this even possible? If we switch to Azure AD the user gets an error logging in as a duplicate user exists, but then login fails.  

More specific question - if we then wanted to delete users from the system, on some users deleting we get this error:

<class 'django.db.utils.IntegrityError'>
update or delete on table "auth_user" violates foreign key constraint "secrets_userkey_user_id_13ada46b_fk_auth_user_id" on table "secrets_userkey"
DETAIL:  Key (id)=(5) is still referenced from table "secrets_userkey".
Python version: 3.10.6
NetBox version: 3.3.4

Wondering what this means and how we would get around this?

Thanks,
Jason

**Comment by a084ed22** [comment]
You likely come from an earlier (2.x) version of netbox that integrated support for secrets. Since 3.x this has moved to be its own plugin and is not part of netbox core.

The database however still references your users' data from the secrets table and doesn't allow you to delete. You could drop the secrets-related tables if you're not making use of the functionality.
